kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PurchaseOrderDocument.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * The Kuali Financial System, a comprehensive financial management system for higher education.
+ *
+ * Copyright 2005-2014 The Kuali Foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package edu.arizona.kfs.module.purap.document;
+
+
+import org.kuali.kfs.sys.businessobject.Building;
+
+import edu.arizona.kfs.sys.businessobject.BuildingExtension;
+
+/**
+ * Purchase Order Document
+ */
+public class PurchaseOrderDocument extends org.kuali.kfs.module.purap.document.PurchaseOrderDocument {
+	private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(PurchaseOrderDocument.class);
+
+    protected String routeCode;
+    protected Building buildingObj;
+
+    /**
+     * Gets the routeCodeObj attribute. 
+     * @return Returns the routeCodeObj.
+     */
+    public Building getBuildingObj() {
+        return buildingObj;
+    }
```

Comment:
getBuildingObj() should be more robust. Refer back to how you coded method BuildingExtension.getRouteCodeObj().